pyproject.toml

```diff
@@ -45,15 +45,42 @@ compatible-mypy = ["mypy>=1.13,<1.16"]
 oracle = ["oracledb"]
 redis = ["redis", "types-redis"]
 
+[dependency-groups]
+tests = [
+    # Dev tools:
+    "pytest==8.3.5",
+    "pytest-mypy-plugins==3.2.0",
+    "pytest-shard==0.1.2",
+    "pytest-xdist==3.7.0",
+    # Django deps:
+    "django==5.2.1",
+    "django-stubs[compatible-mypy,oracle,redis]",
+    "mysqlclient==2.2.7",
+    "psycopg2-binary==2.9.10",
+    "jinja2==3.1.6",
+    "pyyaml==6.0.2",
+    # typing:
+    "mypy==1.15.0"
+]
+pyright = ["pyright==1.1.401"]
+
+dev = [
+    {include-group = "tests"},
+    {include-group = "pyright"},
+]
+
 [project.urls]
 Homepage = "https://github.com/typeddjango/django-stubs"
 Funding = "https://github.com/sponsors/typeddjango"
 "Release notes" = "https://github.com/typeddjango/django-stubs/releases"
 
+[tool.uv.sources]
+django-stubs = { workspace = true }
+django-stubs-ext = { path = "ext", editable = true }
+
 [tool.hatch.build]
 packages = ["django-stubs", "mypy_django_plugin"]
```

Comment:
Do we need to keep `hatch` or not?

Reply:
This part is for the `hatchling` build backend, uv also provides created a build backend but it's not [production ready](https://docs.astral.sh/uv/configuration/build-backend/) yet. 

uv will use hatchling as the default build backend and it's also the recommended option by [packaging.python.org](https://packaging.python.org/en/latest/tutorials/packaging-projects/#choosing-a-build-backend)